### PR TITLE
ARROW-1080: C++: Add tutorial about converting to/from row-wise representation

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

**Comment by xhochy** [comment]
cc @pacman82

File: cpp/apidoc/tutorials/row_wise_conversion.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+// For simplicity, we assume that all arrays consist of a single chunk here.
+// In a productive implementation this should either be explicitly check or code
+// added that can treat chunked arrays.
+
+auto ids = std::static_pointer_cast<arrow::Int64Array>(
+    table->column(0)->data()->chunk(0));
+auto costs = std::static_pointer_cast<arrow::DoubleArray(
+    table->column(1)->data()->chunk(1));
+auto cost_components = std::static_pointer_cast<arrow::ListArray(
+    table->column(2)->data()->chunk(2));
```
Comment (BryanCutler) [inline cpp/apidoc/tutorials/row_wise_conversion.md]: Should this and the above be `->chunk(0)`?

**Review by BryanCutler — COMMENTED** [review]
Thanks for doing this, it looks great!  I just had that one minor question, otherwise LGTM

**Comment by jeffknupp** [comment]
@xhochy are you open to suggestions about grammar / wording on this change?

**Comment by xhochy** [comment]
@jeffknupp As I'm not a native English speaker, I'm very grateful for that ;)

**Review by wesm — APPROVED** [review]
+1

**Comment by wesm** [comment]
I think it's fine to submit follow up patches to expand or improve this part of the docs

**Pull request closed without merging**